restapi/src/main/java/io/stargate/web/resources/v2/schemas/UserDefinedTypesResource.java

```diff
@@ -0,0 +1,374 @@
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import org.apache.cassandra.stargate.db.ConsistencyLevel;
+
+/**
+ * Exposes REST Endpoint to work with Cassandra User Defined Types
+ *
+ * @see https://cassandra.apache.org/doc/latest/cql/types.html
+ * @author Cedrick LUNVEN (@clunven)
+ */
+@Api(
+    produces = APPLICATION_JSON,
+    consumes = APPLICATION_JSON,
+    tags = {"schemas"})
+@Produces(APPLICATION_JSON)
+@Consumes(APPLICATION_JSON)
+@Path("/v2/schemas/keyspaces/{keyspaceName}/types")
+public class UserDefinedTypesResource {
+
+  public static final String HEADER_TOKEN_AUTHENTICATION = "X-Cassandra-Token";
+  public static final String RESOURCE_KEYSPACES = "/v2/schemas/keyspaces/";
+  public static final String PATH_PARAM_KEYSPACE = "keyspaceName";
+
+  /** Reference to the Data Store */
+  @Inject private Db db;
+
+  @Timed
+  @GET
+  @ApiOperation(
+      value = "Get all user defined types (UDT). ",
+      notes = "Retrieve all user defined types (UDT) in a specific keyspace.",
+      response = ResponseWrapper.class,
+      responseContainer = "List")
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 200, message = "OK", response = ResponseWrapper.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 404, message = "Keyspace has not been found", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal server error", response = Error.class)
+      })
+  public Response findAll(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam(HEADER_TOKEN_AUTHENTICATION)
+          String token,
+      @ApiParam(value = "Keyspace to find all udts", required = true)
+          @PathParam(PATH_PARAM_KEYSPACE)
+          final String keyspaceName,
+      @ApiParam(value = "Unwrap results", defaultValue = "false") @QueryParam("raw")
+          final boolean raw,
+      @Context HttpServletRequest request) {
+    return RequestHandler.handle(
+        () -> {
+          // Accessing datastore
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, getAllHeaders(request));
+          // Load User Defined Type Definitions
+          List<UserDefinedTypeResponse> udtResponses =
+              authenticatedDB.getTypes(keyspaceName).stream()
+                  .map(this::mapUdtAsReponse)
+                  .collect(Collectors.toList());
+          // Filter based on user credentials
+          db.getAuthorizationService()
+              .authorizeSchemaRead(
+                  authenticatedDB.getAuthenticationSubject(),
+                  Collections.singletonList(keyspaceName),
+                  udtResponses.stream()
+                      .map(UserDefinedTypeResponse::getName)
+                      .collect(Collectors.toList()),
```

Comment:
Sincere question: wouldn't suffice to check if the user has schema read privilege to the keyspace? There's really a need of passing this list here or could we just pass `null`?

Reply:
Really good question. The RBAC in Cassandra is can be at Table/Type level so having this control would be relevant. 

Now `authorizeSchemaWrite()/authorizeSchemaRead()` should probably perform a filter and not a ok/ko returning only granted objects